docs/book/start/logging-packets.md

```diff
@@ -8,88 +8,18 @@ Let's expand this program to log the traffic that is being permitted in the user
+To log the data in eBPF program, we are going to use aya-log library, which sends the log message and
+the arguments to the user-space program, which then formats and logs the message. This is done in
+order to offload the complexity of parsing and formatting logging messages to user-space - these
+operations are tend to be too complex to be done in eBPF.
 
 While we could go all out and extract data all the way up to L7, we'll constrain our firewall to L3, and to make things easier, IPv4 only.
-The data structure that we'll need to send information to user-space will need to hold an IPv4 address and an action for Permit/Deny, we'll encode both as a `u32`.
-
-```rust linenums="1" title="xdp-log-common/src/lib.rs"
---8<-- "examples/xdp-log/xdp-log-common/src/lib.rs"
-```
-
-1. We implement the `aya::Pod` trait for our struct since it is Plain Old Data as can be safely converted to a byte-slice and back.
-
-!!! tip "Alignment, padding and verifier errors"
-
-    At program load time, the eBPF verifier checks that all the memory used is
-    properly initialized. This can be a problem if - to ensure alignment - the
-    compiler inserts padding bytes between fields in your types.
-
-    **Example:**
-
-    ```rust
-    #[repr(C)]
-    struct SourceInfo {
-        source_port: u16,
-        source_ip: u32,
-    }
-
-    let port = ...;
-    let ip = ...;
-    let si = SourceInfo { source_port: port, source_ip: ip };
-    ```
-
-    In the example above, the compiler will insert two extra bytes between the
-    struct fields `source_port` and `source_ip` to make sure that `source_ip` is
-    correctly aligned to a 4 bytes address (assuming `mem::align_of::<u32>() ==
-    4`).  Since padding bytes are typically not initialized by the compiler,
-    this will result in the infamous `invalid indirect read from stack` verifier
-    error.
-
-    To avoid the error, you can either manually ensure that all the fields in
-    your types are correctly aligned (eg by explicitly adding padding or by
-    making field types larger to enforce alignment) or use `#[repr(packed)]`.
-    Since the latter comes with its own footguns and can perform less
-    efficiently, explicitly adding padding or tweaking alignment is recommended.
-
-    **Solution ensuring alignment using larger types:**
-
-    ```rust
-    #[repr(C)]
-    struct SourceInfo {
-        source_port: u32,
-        source_ip: u32,
-    }
-
-    let port = ...;
-    let ip = ...;
-    let si = SourceInfo { source_port: port, source_ip: ip };
-    ```
-
-    **Solution with explicit padding:**
-
-    ```rust
-    #[repr(C)]
-    struct SourceInfo {
-        source_port: u16,
-        padding: u16,
-        source_ip: u32,
-    }
-
-    let port = ...;
-    let ip = ...;
-    let si = SourceInfo { source_port: port, padding: 0, source_ip: ip };
-    ```
-
-## Writing Data
 
 ### Using Kernel Network Types
```

Comment:
Again, we should never mention the kernel at all in a networking howto. This is standard networking structures. No need to mention `ethhdr` etc.